compiled_starters/sqlite-starter-go/db.go

```diff
@@ -0,0 +1,67 @@
+package sqlite
+
+import (
+	"fmt"
+	"os"
+
+	"github/com/codecrafters-io/sqlite-starter-go/dbfile"
+)
+
+type (
+	DB struct {
+		f *os.File
+
+		dbHeader dbfile.DBHeader
+
+		pageSize int64
+	}
+
+	Info struct {
+		Tables int
+	}
+)
+
+func Open(file string) (*DB, error) {
+	f, err := os.OpenFile(file, os.O_RDONLY, 0644)
+	if err != nil {
+		return nil, fmt.Errorf("open file: %w", err)
+	}
+
+	h, err := dbfile.ReadDBHeader(f)
+	if err != nil {
+		return nil, fmt.Errorf("read db header: %w", err)
+	}
+
+	if h.Magic() != dbfile.Magic {
+		return nil, fmt.Errorf("not an sqlite3 database")
+	}
+
+	d := &DB{
+		f:        f,
+		pageSize: h.PageSize(),
```

Comment:
I like to have frequently used values near in the struct. That way it's easier to test a bunch of the code, for example. We might want to not to store the header, or we might want to change how we store it. It's constant for opened DB, so we won't have problems with synchronisation. Instead of calling function on a field we just use field, which is a bit cleaner to me.

Should we remove the pageSize?